text/0162-refactoring-support.md

```diff
@@ -0,0 +1,622 @@
+     3/3 | 2:03:20 PM | REFACTOR_COMPLETE    | AWS::Lambda::Function         | MyStack.FunctionA5EA2BD8     | Service.FunctionA5EA2BD8     
+    
+    ✅  Stack refactor complete
+
+You can configure the refactoring behavior by passing the `--refactoring` flag
+to the CLI, or by configuring this setting in the `cdk.json` file:
+
+```json
+{
+  "app": "...",
+  "refactoring": "EXECUTE_AND_DEPLOY"
+}
+```
+
+Please note that the same CDK application can have multiple stacks for different
+environments. In that case, the CLI will group the stacks by environment and
+perform the refactoring separately in each one. Trying to move resources between
+stacks that belong in different environments will result in an error.
+
+### Rollbacks
+
+After refactoring the stack, the CLI will proceed with the deployment
+(assuming that is your choice). If the deployment fails, and CloudFormation
+rolls it back, the CLI will execute a second refactor, in reverse, to bring the
+resources back to their original locations.
+
+### Ambiguity
+
+Imagine a person walking down the street, and someone takes two snapshots of
+them, a few seconds apart. When you look at the snapshots, you can tell that
+it's the _same person_, but at different locations, and not two different
+people. What leads you this conclusion? The fact that the person's features
+(face, clothes, height, etc.) are exactly the same in both photographs. But if,
+instead of one person walking, you have two identical siblings, wearing the same
+clothes, you won't be able to tell which one is which, from one photo to the
+next.
+
+The same principle applies here: you can think of stack name + logical ID as a
+place, and the resource properties as its "personal" features. If a resource has
+different properties before and after a potential deployment, the best we can do
+is assume they are different resources. If the properties are the same from one
+point in time to the other, they very likely are the same resource (although the
+developer still has the last word on this). But if there are two resources that
+have the same properties in the same stack, they are like the twin siblings
+case: there's no way to tell them which is which in case they both move to a
+different place.
+
+Indistinguishable resources in this sense are said to be _equivalent_ (see
+Appendix A for a more formal definition). In the unlikely event that two or more
+equivalent resources move, the CLI won't be able to proceed. For example,
+suppose you have two queues in the same stack, named `Queue1` and `Queue2`, with
+the same properties, and without a hard-coded physical ID:
+
+    App
+    └ Stack
+      ├ Queue1
+      └ Queue2
+
+If they get renamed to, let's say, `Queue3` and `Queue4`,
+
+    App
+    └ Stack
+      ├ Queue3
+      └ Queue4
+
+then the CLI will not be able to establish a 1:1 mapping between the old and new
+names. In this case, it will show you the ambiguity, and stop the deployment:
```

Comment:
I still think we should allow the user to provide a resolution file in this case (or any case really) that explicitly maps these changes.